### DISCUSSION TITLE
Can I use datasets that contain multiple different molecules (i.e. different number of atoms)?

### DISCUSSION THREAD

**Discussion by jcartus** [mir-group/nequip#200] [Q&A] [answered]
I would like to use NeqIP to predict the energy (and maybe forces) in monolayers of various molecules. The molecules are arranged in several different unit cells. As a result the number of atoms are different for each point in my data set. 

After studying the examples in the repo and the paper I am under the impression that this is not possible. Yet [in this question](https://github.com/mir-group/nequip/discussions/137) it seems a user has applied NeqIP to a data set with different numbers of iron atoms per unit cell. 

So does it work after all and did I just misinterpret the examples/documentaion? Thanks!

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @jcartus , 

This is definitely possible, just not with the `NpzDataset`, which we use in `example.yaml` and `minimal.yaml` because the data is already in that format. We recommend converting your dataset into the `extxyz` format and using the ASE dataset feature:
```
dataset: ase
dataset_file_name: whatever.xyz
ase_args:
  format: extxyz
```
You can also use any other format that ASE can read and change `format` accordingly.

Please see `full.yaml` for more details: https://github.com/mir-group/nequip/blob/main/configs/full.yaml#L69-L95

**Reply by jcartus** [reply]
That's great to hear! Thank you very much for the super quick reply :)

**Comment by Linux-cpp-lisp** [comment]
(Just a note on what you are trying to do: even if you are only interested in energy predictions, if you have force data in your training data you should definitely train with both energies and forces. The training will be much faster, and likely more accurate.  If you only need energy predictions without the overhead of force computation, you can then later disable the force computation during inference if performance is important for you.)
